### DISCUSSION TITLE
카톡 버전 몇으로 쓰시나요?

### DISCUSSION THREAD

**Discussion by kiwioverflow** [storycraft/node-kakao#696] [General]
최근에 특정 카톡 버전에서 몇일 이상 로그인 상태를 유지하면 영구정지를 불러오는 듯한 느낌입니다.

다들 카톡 몇 버전 (ex. 3.3.3.2917)으로 사용하시고 계신가요?

**Comment by mschadev** [comment]
`3.3.3.2911` 씁니다.

**Reply by kiwioverflow** [reply]
저도 해당버전 쓰다가 PC인증된 날짜 기준으로 일주일 안쪽으로 해외계정, 국내계정 둘다 영정을 먹던데..

따로 영정이나 그런 부분은 없으셨나요?

**Reply by mschadev** [reply]
@parkkw472 없었습니다.

**Comment by benesiaxd** [comment]
???

**Reply by benesiaxd** [reply]
kickuser 부분을 보세요.
요새 배포업자들 message내에 userid를 변조해서 보내기때문에
data.chat.sender로 강퇴하시면안됩니다

**Reply by benesiaxd** [reply]
상식적으로 로그인해있다고 정지는 아니죠

**Reply by tadashi74** [reply]
> kickuser 부분을 보세요. 요새 배포업자들 message내에 userid를 변조해서 보내기때문에 data.chat.sender로 강퇴하시면안됩니다

그러면 어떤걸로 내보내기 처리해야 되는지 여쭤봐도 될까요?

**Reply by benesiaxd** [reply]
getsenderinfo

**Reply by tadashi74** [reply]
> getsenderinfo

감사합니다 :)

**Reply by kiwioverflow** [reply]
> 상식적으로 로그인해있다고 정지는 아니죠

@benesiaxd
![zIgXSYufzI](https://user-images.githubusercontent.com/39560346/135797796-4e311b43-b4d5-4780-990e-98e78d2970cc.png)
로그인해두고 방치한지 약 1주일정도 지난 국내 010 계정입니다.

![image](https://user-images.githubusercontent.com/39560346/135797937-b8c4273e-f423-43a8-94d6-c499294c7a78.png)
이것도 1주일정도 지난 해외계정입니다.

서로 버전도 다른데 왜 일주일 안쪽으로 정지를 먹는건지 모르겠네요
가장 의심되는 부분은 XVC 값이 해당 버전과 맞지 않아서 그런것같기도 하고요.
접속 위치는 국내 IP입니다.

**Reply by disabled-variable** [reply]
> kickuser 부분을 보세요. 요새 배포업자들 message내에 userid를 변조해서 보내기때문에 data.chat.sender로 강퇴하시면안됩니다

userid 변조가 가능한가요? ㄷㄷ

**Comment by generalxd** [comment]
저는 노드카카오 4.4.0 버전 기본 설정버전씁니다.

3.2.3.2698 이네요

계정 30개 하나도 정지먹지않았습니다.

가리기를 너무 빠르게 한다던지 하는 이유가 누적된거 아닐까요 ?

**Reply by kiwioverflow** [reply]
> 가리기를 너무 빠르게 한다던지 하는 이유가 누적된거 아닐까요 ?

가리기의 경우 Rate Limit을 적용시켜 초당 요청수를 10회로 제한시켜놓은 상태입니다.

봇을 사용 안하고 있을 때에도 정지를 먹는것으로 보아, 요청 횟수와는 약간 거리가 멀지 않나 싶습니다

**Reply by generalxd** [reply]
초당 10개는 빠릅니다. 무적 계정이 되면야 상관없는데 그 전 단계면 초당 8개 정도가 한계입니다

**Reply by kiwioverflow** [reply]
어짜피 10개 이상으로 넘어가면 -303 응답으로 카톡에서 역으로 속도제한을 걸어버리더라고요 ㅋㅋㅋ

그래도 딱히 정지 안먹어서 내비두고 쓰고 있었는데 흠

일단 한번 참고해서 테스트를 해봐야 할거같네요

감사합니다

**Comment by generalxd** [comment]
혹시 V3 버전 쓰시나요?

**Reply by kiwioverflow** [reply]
v3버전에서 v4이슈 참고하여 일부 값 수정하면서 사용하고 있었습니다

혹시 v3버전에서 정지먹을 만한 이슈가 있을까요?

**Reply by generalxd** [reply]
V3 버전은 기존 버전과 다른점이 매우 많습니다.

V3 버전 영구정지는 예전부터 흔했습니다. V4로 넘어오세요

**Reply by kiwioverflow** [reply]
안그래도 v4 마이그레이션 진행중이었는데 오늘 내일중으로 넘어가야겠네요

감사합니다!
